packages/amplify-graphql-model-transformer/src/graphql-model-transformer.ts

```diff
@@ -398,13 +428,15 @@ export class ModelTransformer extends TransformerModelBase implements Transforme
             default:
               throw new Error('Unknown subscription field type');
           }
-          resolver.addToSlot(
-            'postAuth',
-            MappingTemplate.s3MappingTemplateFromString(
-              generateAuthExpressionForSandboxMode(context.sandboxModeEnabled),
-              `${subscription.typeName}.${subscription.fieldName}.{slotName}.{slotIndex}.req.vtl`,
-            ),
-          );
+          if (subscriptionLevel === SubscriptionLevel.on) {
```

Comment:
is there a reason we're checking `equals to on` here and in the above `if` conditional is changed to check for `not equals to off`? Isn't this always `on` based on the above check?

Reply:
public is also an option in addition to off/on, which is why I specify to avoid getting authorization on those public subscriptions